src/db/group.rs

```diff
@@ -148,6 +157,156 @@ impl Group {
     pub fn get_expiry_time(&self) -> Option<&chrono::NaiveDateTime> {
         self.times.get_expiry()
     }
+
+    pub fn entries(&self) -> Vec<&Entry> {
+        let mut response: Vec<&Entry> = vec![];
+        for node in &self.children {
+            if let Node::Entry(e) = node {
+                response.push(e)
+            }
+        }
+        response
+    }
+
+    pub fn entries_mut(&mut self) -> Vec<&mut Entry> {
+        let mut response: Vec<&mut Entry> = vec![];
+        for node in &mut self.children {
+            if let Node::Entry(e) = node {
+                response.push(e)
+            }
+        }
+        response
+    }
+
+    pub fn groups_mut(&mut self) -> Vec<&mut Group> {
+        let mut response: Vec<&mut Group> = vec![];
+        for node in &mut self.children {
+            if let Node::Group(g) = node {
+                response.push(g);
+            }
+        }
+        response
+    }
+
+    fn replace_entry(&mut self, entry: &Entry) {
+        for node in &mut self.children {
+            match node {
+                Node::Group(g) => {
+                    g.replace_entry(entry);
+                }
+                Node::Entry(e) => {
+                    if e.uuid == entry.uuid {
+                        *e = entry.clone();
+                    }
+                }
+            }
+        }
+    }
+
+    pub fn find_entry_by_uuid(&self, id: Uuid) -> Option<&Entry> {
+        for node in &self.children {
+            match node {
+                Node::Group(g) => {
+                    if let Some(e) = g.find_entry_by_uuid(id) {
+                        return Some(e);
+                    }
+                }
+                Node::Entry(e) => {
+                    if e.uuid == id {
+                        return Some(e);
+                    }
+                }
+            }
+        }
+        None
```

Comment:
You should probably implement a recursive iterator, so you can do this:

```suggestion
        self.iter_recursive()
            .flat_map(|c| {
                match c {
                    Node::Entry(e) => Some(e),
                    _ => None
                }
            })
            .find(|e| e.uuid == id)
```